test/deal-investor/post-deal-investors.api-test.ts

```diff
@@ -0,0 +1,309 @@
+import { PROPERTIES } from '@ukef/constants';
+import { UkefId } from '@ukef/helpers';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { AcbsCreateDealInvestorRequest } from '@ukef/modules/deal-investor/dto/acbs-create-deal-investor-request.dto';
+import { withAcbsAuthenticationApiTests } from '@ukef-test/common-tests/acbs-authentication-api-tests';
+import { IncorrectAuthArg, withClientAuthenticationTests } from '@ukef-test/common-tests/client-authentication-api-tests';
+import { withDateOnlyFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/date-only-field-validation-api-tests';
+import { withStringFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/string-field-validation-api-tests';
+import { Api } from '@ukef-test/support/api';
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { CreateDealInvestorGenerator } from '@ukef-test/support/generator/create-deal-investor-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+describe('POST /deals/{dealIdentifier}/investors', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+
+  const dealIdentifier: UkefId = valueGenerator.ukefId();
+  const createDealInvestorUrl = `/api/v1/deals/${dealIdentifier}/investors`;
+
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+
+  const { acbsRequestBodyToCreateDealInvestor, requestBodyToCreateDealInvestor } = new CreateDealInvestorGenerator(
+    valueGenerator,
+    dateStringTransformations,
+  ).generate({
+    numberToGenerate: 2,
```

Comment:
Why do we generate 2?

Reply:
As above, I was thinking it was good to check do tests with arrays containing multiple elements (as such input is valid) just so we can be certain that the logic of choosing the first element in the array etc. works fine, but I'm happy to change this

Reply:
That makes sense, I'm happy for you to keep it